src/CodeGen_GPU_Dev.cpp

```diff
@@ -94,5 +97,55 @@ bool CodeGen_GPU_Dev::is_buffer_constant(const Stmt &kernel,
     return v.result;
 }
 
+namespace {
+
+class ScalarizePredicatedLoadStore : public IRMutator {
+public:
+    using IRMutator::mutate;
+    using IRMutator::visit;
+
+protected:
+    Stmt visit(const Store *s) override {
+        if (!is_const_one(s->predicate)) {
+            std::vector<Stmt> scalar_stmts;
+            for (int ln = 0; ln < s->value.type().lanes(); ln++) {
+                scalar_stmts.push_back(IfThenElse::make(
+                                           extract_lane(s->predicate, ln),
```

Comment:
It *might* be important to detect predicates of the form ramp < broadcast, and generate a nested if in that case. That is what the old logic did (implicitly/indirectly). I think this is simpler though, so we should see how this performs first.

Reply:
That's a good point.  I'm not sure whether there will be perf differences on other backends, but my limited testing on Metal seems to indicate about the same performance.  Happy to revisit if anyone finds performance issues.